src/NHSD.BuyingCatalogue.Ordering.Api/Infrastructure/ObjectExtensions.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System;
+
+namespace NHSD.BuyingCatalogue.Ordering.Api.Infrastructure
+{
+    public static  class ObjectExtensions
```

Comment:
In one of the previous code standards meetings we have decided against using the extensions throw if null. Sorry that the documentation isnt written up yet.

Reply:
@RuthBerry96 is correct – in a previous standards meeting we decided to abandon the `ThrowIfNull` extension method. The following considerations were given.

**For**

- DRY

**Against**

- While technically/syntactically fine, it is semantically incorrect
- There is something a bit smelly about extension methods that essentially extend everything (see this [response](https://stackoverflow.com/a/7652359) from Eric Lippert)
- Without the extension method the call stack will show the true root of the exception (assuming no inlining of the extension method)
- With the extension there was a tendency to omit the parameter name
- The throw expressions introduced in C#7 make assignment statements one liners now anyway

Therefore, we can omit the `ObjectExtensions` class.

Hopefully I will manage to document some of these decisions next sprint...

Reply:
are we saying use  the following 

var request = httpContext?.Request ?? throw new ArgumentNullException(nameof(httpContext));

instead of the extension method or just drop the null check all together ?

Reply:
```c#
var request = httpContext?.Request ?? throw new ArgumentNullException(nameof(httpContext));
```